frontend/src/components/Town/interactables/Chess/GameReviewDetail.tsx

```diff
@@ -0,0 +1,276 @@
+    g7: { pieceType: 'P', pieceColor: 'B', moved: false },
+    h7: { pieceType: 'P', pieceColor: 'B', moved: false },
+    a8: { pieceType: 'R', pieceColor: 'B', moved: false },
+    b8: { pieceType: 'N', pieceColor: 'B', moved: false },
+    c8: { pieceType: 'B', pieceColor: 'B', moved: false },
+    d8: { pieceType: 'Q', pieceColor: 'B', moved: false },
+    e8: { pieceType: 'K', pieceColor: 'B', moved: false },
+    f8: { pieceType: 'B', pieceColor: 'B', moved: false },
+    g8: { pieceType: 'N', pieceColor: 'B', moved: false },
+    h8: { pieceType: 'R', pieceColor: 'B', moved: false },
+  };
+  Object.entries(initialPositions).forEach(([key, piece]) => {
+    const file = key[0] as ChessFilePosition;
+    const rank = parseInt(key[1], 10) as ChessRankPosition;
+    if (piece) {
+      const cell: ChessCell = {
+        piece,
+      };
+      board[rankToIndex(rank)][fileToIndex(file)] = cell;
+    }
+  });
+  return board;
+};
+
+const pieceTypeFromFENChar: { [key: string]: ChessPiece['pieceType'] } = {
+  K: 'K',
+  Q: 'Q',
+  R: 'R',
+  B: 'B',
+  N: 'N',
+  P: 'P',
+  k: 'K',
+  q: 'Q',
+  r: 'R',
+  b: 'B',
+  n: 'N',
+  p: 'P',
+};
+
+const parseFEN = (fen: string) => {
+  const [position, turn, castling, enPassant, halfMoveClock, fullMoveNumber] = fen.split(' ');
+  const rows = position.split('/');
+  const board: ChessCell[][] = rows.map(row => {
+    const rowPieces: ChessCell[] = [];
+    for (const char of row) {
+      if (isNaN(parseInt(char))) {
+        const pieceColor = char === char.toUpperCase() ? 'W' : 'B';
+        const pieceType = pieceTypeFromFENChar[char];
+        rowPieces.push({ piece: { pieceType, pieceColor, moved: false } });
+      } else {
+        rowPieces.push(...Array(parseInt(char)).fill(undefined));
+      }
+    }
+    return rowPieces;
+  });
+
+  return {
+    board,
+    turn,
+    castling,
+    enPassant,
+    halfMoveClock: parseInt(halfMoveClock),
+    fullMoveNumber: parseInt(fullMoveNumber),
+  };
+};
+
+const applyFENToBoard = (fen: string) => {
+  const { board, turn, castling, enPassant, halfMoveClock, fullMoveNumber } = parseFEN(fen);
+
+  return { board, turn, castling, enPassant, halfMoveClock, fullMoveNumber };
+};
+
+export default function GameReviewDetail(props: GameReviewDetailProps): JSX.Element {
+  const { game, mainMenu, backToReviewList } = props;
+  const [currentMoveIndex, setCurrentMoveIndex] = useState(0);
+
+  // Parse the moves once and store in a state
+  const [parsedMoves, setParsedMoves] = useState<string[]>([]);
+
+  useEffect(() => {
+    // Check if game.moves is a valid JSON string
+    if (game.moves && typeof game.moves === 'string') {
+      try {
+        const movesArray = JSON.parse(game.moves);
+        setParsedMoves(movesArray);
+      } catch (error) {
+        console.error('Error parsing game moves:', error);
+      }
+    }
+  }, [game.moves]);
+
+  const [gameState, setGameState] = useState({
```

Comment:
The current use of the fen notation to display the board is well done. But for the move displays (previous, current, and next) move could we show the moves using the chess coordinates. I think that the actual fen would not be understood by a typical user.

Reply:
I agree this is def something I was thinking about before, and I think the proposed solution you send me before works